### PR TITLE
Feat/pm2022 008

### PR DESCRIPTION
**Pull request by rifisdfds**
This is a PR for the Kyverno post-mortem

### CODE REVIEW

File: PM2022-008 - CRUD operations on K8s Service resources unavailable.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+All times in UTC.
+
+| Time                   | Event                                                                                                         |
+| :--------------------- | ------------------------------------------------------------------------------------------------------------- |
+| 2022-08-03 11:38       | Alerted by end-user team that they were having issues applying a Service resource to the affected K8s cluster |
+| 2022-08-03 11:50       | Started looking into the alerted issue                                                                        |
+| 2022-08-03 12:08       | Issue identified, started work on fix                                                                         |
+| 2022-08-03 12:18       | Issue recreated in K8s sandbox                                                                                |
+| 2022-08-03 12:26-12:?? | Fix deployed in K8s sandbox                                                                                   |
```
[the remark this answers is not in the document]
Comment (rifisdfds) [reply PM2022-008 - CRUD operations on K8s Service resources unavailable.md:17]: Did not record time. It was after 12:26 and before the next event, hence the ??

File: PM2022-008 - CRUD operations on K8s Service resources unavailable.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+## Contributing Factors
+
+- Kyverno not being kept up-to-date due to code freeze
+- Issue with certificate chain expiry in third party application through no action of our own
+- Signs of Kyverno not being a mature project
+- If the Kyverno ClusterPolicy's 'failurePolicy' was set to 'ignore' instead of the default 'fail', the issue wouldn't have occurred
+- Issue raised but not on the 'ops day to day' board
+- Product put into production prematurely without monitoring
+- Not enough communication and coordination before product put into production
```
Comment (jdsmithit) [inline PM2022-008 - CRUD operations on K8s Service resources unavailable.md:29]: - Not enough communication and coordination before **the** product was put into production

File: PM2022-008 - CRUD operations on K8s Service resources unavailable.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+## Action Items
+
+- Implement monitoring on Kyverno
+- Create internal documentation on our Kyverno setup
+- Investigate the exact issue with certificate in the application
```
Comment (jdsmithit) [inline PM2022-008 - CRUD operations on K8s Service resources unavailable.md:35]: - Investigate the exact issue with the certificate in the application

File: PM2022-008 - CRUD operations on K8s Service resources unavailable.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+CRUD operations on any K8s Service resource excluding exempted namespaces were unavailable due to an unresponsive webhook. More specifically, Kyverno was stuck in crash loop. This was resolved by updating to a newer Helm Chart version of Kyverno.
+
+## Timeline
+
+All times in UTC.
+
+| Time                   | Event                                                                                                         |
+| :--------------------- | ------------------------------------------------------------------------------------------------------------- |
+| 2022-08-03 11:38       | Alerted by end-user team that they were having issues applying a Service resource to the affected K8s cluster |
```
Comment (jdsmithit) [inline PM2022-008 - CRUD operations on K8s Service resources unavailable.md:13]: Alerted by an end-user team

File: PM2022-008 - CRUD operations on K8s Service resources unavailable.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
+# CRUD operations on K8s Service resources unavailable
+
+## Summary
+
+CRUD operations on any K8s Service resource excluding exempted namespaces were unavailable due to an unresponsive webhook. More specifically, Kyverno was stuck in crash loop. This was resolved by updating to a newer Helm Chart version of Kyverno.
```
Comment (jdsmithit) [inline PM2022-008 - CRUD operations on K8s Service resources unavailable.md:5]: Kyverno was stuck in a crash

File: PM2022-008 - CRUD operations on K8s Service resources unavailable.md (hunk 1 above)
Comment (jdsmithit) [inline PM2022-008 - CRUD operations on K8s Service resources unavailable.md:17]: is the ?? left in on purpose

**Review by jdsmithit — APPROVED** [review]

**Pull request merged**